solutionbox/structured_data/datalab_solutions/structured_data/trainer/task.py

```diff
@@ -0,0 +1,357 @@
+
+  problem_type = transform_config['problem_type']
+  if problem_type != 'regression' and problem_type != 'classification':
+    print('ERROR: problem_type from the transform file should be regression'
+          ' or classification.')
+    sys.exit(1)
+
+  model_type = transform_config['model_type']
+
+  if model_type != 'dnn' and model_type != 'linear':
+    print('ERROR: model_type from the transform file should be dnn or linear')
+    sys.exit(1)
+
+  # Set how often to run checkpointing in terms of time.
+  config = tf.contrib.learn.RunConfig(
+      save_checkpoints_secs=args.save_checkpoints_secs)
+
+  # TODO(brandondutra) check layer_sizes pass in if needed.
+
+  if problem_type == 'regression' and model_type == 'dnn':
+    estimator = tf.contrib.learn.DNNRegressor(
+        feature_columns=feature_columns,
+        hidden_units=args.layer_sizes,
+        config=config,
+        model_dir=train_dir,
+        optimizer=tf.train.AdamOptimizer(
+          args.learning_rate, epsilon=args.epsilon))
+  elif problem_type == 'regression' and model_type == 'linear':
+    estimator = tf.contrib.learn.LinearRegressor(
+        feature_columns=feature_columns,
+        config=config,
+        model_dir=train_dir,
+        optimizer=tf.train.AdamOptimizer(
+          args.learning_rate, epsilon=args.epsilon))
+  elif problem_type == 'classification' and model_type == 'dnn':
+    estimator = tf.contrib.learn.DNNClassifier(
+        feature_columns=feature_columns,
+        hidden_units=args.layer_sizes,
+        n_classes=metadata.stats['labels'],
+        config=config,
+        model_dir=train_dir,
+        optimizer=tf.train.AdamOptimizer(
+          args.learning_rate, epsilon=args.epsilon))
+  elif problem_type == 'classification' and model_type == 'linear':
+    estimator = tf.contrib.learn.LinearClassifier(
+        feature_columns=feature_columns,
+        n_classes=metadata.stats['labels'],
+        config=config,
+        model_dir=train_dir,
+        optimizer=tf.train.AdamOptimizer(
+          args.learning_rate, epsilon=args.epsilon))
+
+  return estimator
+
+
+def get_experiment_fn(args):
+  """Builds the experiment function for learn_runner.run"""
+
+  def get_experiment(output_dir):
+
+    # Load the metadata.
+    metadata = ml.features.FeatureMetadata.get_metadata(
+        args.metadata_path)
+
+    # Load the config file.
+    transform_config = json.loads(
+        ml.util._file.load_file(args.transforms_config_file))
+
+    # Build tf.learn features
+    feature_columns = util.produce_feature_columns(metadata, transform_config)
+
+    # Get the model to train.
+    estimator = get_estimator(output_dir, feature_columns, metadata, 
+                              transform_config, args)
+
+    input_placeholder_for_prediction = get_placeholder_input_fn(metadata, 
+        transform_config)
+
+    # Save the finished model to output_dir/model
+    export_monitor = util.ExportLastModelMonitor(
```

Comment:
Yes. "additional_assets=[args.metadata_path]" puts a copy of the metadata.json file in the final model dir.